21 a view for tasks associated with an user

liber09 opened the pull request.

ved0 commented on client/js/userTaskView.js:
```diff
@@ -0,0 +1,185 @@
+            // Assuming user information is stored in the task data
+            return selectedUser !== null ? (selectedUser === '' ? !task.user : task.user === selectedUser) : true;
+        });
+
```
Here we could make it a bit more interactive , so when the user selects an item from the list I feel like the text should also change to reflect that. For example,
Adding, on like line 15:
`const userDropdownButton = document.getElementById('userDropdown');`

and here each time the selectUser function is called:
`if (selectedUser === null || selectedUser === '') {
userDropdownButton.textContent = selectedUser === null ? 'All users' : 'Unassigned';
} else {
userDropdownButton.textContent = selectedUser;
 }
`

ved0 commented on client/js/userTaskView.js:
```diff
@@ -0,0 +1,185 @@
+        thClicked.classList.toggle('ascending', isAscending);
+        thClicked.classList.toggle('descending', !isAscending);
+
+        populateTable(data);
```
When you send it in like this, you are sending in `selectedUser = undefined`.  So when you try to sort it, it displays that it cant find tasks for user "undefined". When you leave and come back to the same user, then it displays a sorted task list.

==> This is relevant if you implement the suggestion down below with the changing text button:
```
        let whatToDisplay = userDropdownButton.textContent === 'All users' ? null : userDropdownButton.textContent === '' ? '':userDropdownButton.textContent;
        populateTable(data, whatToDisplay);
```

Otherwise, you need to handle undefined in the populateTable method :)

liber09 replied:
Sorting bug is now fixed

ved0 reviewed (changes requested):
I think that you did a great job. 👍 
I suggested some cosmetic improvements, minor ones. 

One change I would like to see before merging to main is fixing the sorting. 

Keep up the good work :)

davidlidebrandt commented:
Is there an easy way of viewing the page/view? I tried copying the folders to a new project and viewed it with live server in VS code, but I'm having some trouble with "MIME type ('text/html') is not a supported", I'm guessing it has something to do with the paths to the linked CSS and JS files.

Also, I think it might be a good idea to put the files from external libraries in a separate folder.

Rasmus3liasson commented:
After adding the code from this pull request and setting up a controller route to test it, the UI seem to match the design.

I did notice, however, that the HTML file is currently placed in the `js` folder. Thinking this might not be the most optimal setup.

I'd like to suggest an alternative approach that could potentially improve our project structure. While I can't say I'm absolutely certain about the best practices in Spring Boot applications, from what I've research, a common approach is to serve static content from the `resources/static` folder. Where we can use a `templates` folder for our html file. This adjustment has the potential to simplify our integration with Thymeleaf, making it easier to serve static content.

liber09 commented:
> Is there an easy way of viewing the page/view? I tried copying the folders to a new project and viewed it with live server in VS code, but I'm having some trouble with "MIME type ('text/html') is not a supported", I'm guessing it has something to do with the paths to the linked CSS and JS files.
> 
> Also, I think it might be a good idea to put the files from external libraries in a separate folder.

I just change the run/debug configuration and select my html file and then press run in intelli J, that works fine for me.

liber09 commented:
> After adding the code from this pull request and setting up a controller route to test it, the UI seem to match the design.
> 
> I did notice, however, that the HTML file is currently placed in the `js` folder. Thinking this might not be the most optimal setup.
> 
> I'd like to suggest an alternative approach that could potentially improve our project structure. While I can't say I'm absolutely certain about the best practices in Spring Boot applications, from what I've research, a common approach is to serve static content from the `resources/static` folder. Where we can use a `templates` folder for our html file. This adjustment has the potential to simplify our integration with Thymeleaf, making it easier to serve static content.

I tried to find best practices but that was not so easy. I found some saying resources/static folder so i have renamed the file to index.,html and moved it to resources/static folder

3xer commented on client/js/userTaskView.js:
```diff
@@ -0,0 +1,193 @@
+
+        // Add "All Users" option
+        const allUsersOption = document.createElement('li');
+        const allUsersAnchor = document.createElement('a');
```
The createElements could be global like this if i remember correct: `const createLiElement = () => document.createElement('li')`, and write `createLiElement();` instead of writing `document.createElement('li'`) or similar for each line where you need it. The code will also be easier to change if you would like to use another element in the future.

3xer reviewed (approved):
left a comment in the code nothing wrong in the code just some possible structure changes otherwise it looks good.

ved0 reviewed (approved):
Great work! 👍
